src/main.tsx

```diff
@@ -22,7 +25,11 @@ const rootElement = document.getElementById('root');
 if (rootElement) {
   createRoot(rootElement).render(
     <StrictMode>
-      <RouterProvider router={router} />
+      <ThemeProvider>
+        <Provider store={store}>
```

Comment:
this provider is enough
it works for whole app
no need use Provider again in components